src/main/java/hudson/tasks/junit/TestResultAction.java

```diff
@@ -223,11 +239,44 @@ public List<CaseResult> getSkippedTests() {
         return getResult().getSkippedTests();
     }
 
+    private TestResult parseOnly() {
+        XmlFile df = getDataFile();
+        TestResult r;
+        try {
+            r = new TestResult();
+            r.parse(df);
+        } catch (Exception e) {
+            logger.log(Level.WARNING, "Failed to load " + df, e);
+            r = new TestResult();   // return a dummy
+        }
+        return r;
+    }
+
+    static ConcurrentHashMap<String, SoftReference<TestResult>> resultCache = new ConcurrentHashMap<>();
```

Comment:
This is incorrect FTR: Jenkins plugins should never keep state in `static` fields. Rather use instance fields of some `@Extension`. Noticed because of a test failure in https://github.com/jenkinsci/parallel-test-executor-plugin/pull/296 where we try to simulate a Jenkins restart without using `RealJenkinsRule` and thus a new JVM.

Reply:
In practice this mainly causes trouble for functional tests. If there is not already an `@Extension` which would be a natural home for the state, you can simply add one

```java
@Extension public static final class TheCache {
    static TheCache get() {
        return ExtensionList.lookupSingleton(TheCache.class);
    }
    private final Cache<Whatever, Else> data = Caffeine.newCache().orSomething();
    // etc.
}
```